lib/optimizely/decision_service.rb

```diff
@@ -159,54 +159,35 @@ def get_variation_for_feature_experiment(feature_flag, user_id, attributes = nil
 
       feature_flag_key = feature_flag['key']
       unless feature_flag['experimentIds'].empty?
-        # check if experiment is part of mutex group
-        experiment_id = feature_flag['experimentIds'][0]
-        experiment = @config.experiment_id_map[experiment_id]
-        unless experiment
-          @config.logger.log(
-            Logger::DEBUG,
-            "Feature flag experiment with ID '#{experiment_id}' is not in the datafile."
-          )
-          return nil
-        end
-
-        group_id = experiment['groupId']
-        # if experiment is part of mutex group we first determine which experiment (if any) in the group the user is part of
-        if group_id and @config.group_key_map.has_key?(group_id)
-          group = @config.group_key_map[group_id]
-          bucketed_experiment_id = @bucketer.find_bucket(user_id, group_id, group['trafficAllocation'])
-          if bucketed_experiment_id.nil?
+        #Evaluate each experiment id and return the first bucketed experiment variation
+        feature_flag['experimentIds'].each do |experiment_id|
+          # check if experiment is part of mutex group
+          experiment = @config.experiment_id_map[experiment_id]
+          unless experiment
             @config.logger.log(
-              Logger::INFO,
-              "The user '#{user_id}' is not bucketed into any of the experiments on the feature '#{feature_flag_key}'."
+                Logger::DEBUG,
+                "Feature flag experiment with ID '#{experiment_id}' is not in the datafile."
             )
             return nil
           end
-        else
-          bucketed_experiment_id = experiment_id
-        end
-
-        if feature_flag['experimentIds'].include?(bucketed_experiment_id)
-          experiment = @config.experiment_id_map[bucketed_experiment_id]
           experiment_key = experiment['key']
           variation_id = get_variation(experiment_key, user_id, attributes)
           unless variation_id.nil?
             variation = @config.variation_id_map[experiment_key][variation_id]
             @config.logger.log(
-              Logger::INFO,
-              "The user '#{user_id}' is bucketed into experiment '#{experiment_key}' of feature '#{feature_flag_key}'."
+                Logger::INFO,
+                "The user '#{user_id}' is bucketed into experiment '#{experiment_key}' of feature '#{feature_flag_key}'."
             )
             return {
-              'variation' => variation,
-              'experiment' => experiment
+                'variation' => variation,
+                'experiment' => experiment
```

Comment:
Please keep experiment first then variation. Should be consistent with method doc. And eases code readability